integration/src/test/resources/non-blocking-test-config-global-err.xml

```diff
@@ -255,16 +255,72 @@
 
     <flow name="foreach" initialState="stopped">
         <test:assert-intercepting responseSameTask="false"/>
+        <test:non-blocking-processor/>
         <foreach>
-            <test:assert-intercepting responseSameTask="false" count="2"/>
+            <test:assert-intercepting responseSameTask="false"/>
             <test:non-blocking-processor/>
             <foreach>
+                <test:assert-intercepting responseSameTask="false"/>
+                <http:request config-ref="http-request-config" path="/test" method="POST" >
+                    <http:response-validator >
+                        <http:success-status-code-validator values="0..599" />
+                    </http:response-validator>
+                </http:request>
+            </foreach>
+        </foreach>
+    </flow>
+
+    <http:request-config name="http-request-config">
+        <http:request-connection host="localhost" port="${requesterPort}"/>
+    </http:request-config>
+
+    <flow name="untilSuccessful" initialState="stopped">
+        <test:assert-intercepting responseSameTask="false"/>
+        <until-successful>
+            <test:assert-intercepting responseSameTask="false" count="2"/>
+            <test:non-blocking-processor/>
+            <until-successful>
                 <test:assert-intercepting responseSameTask="false" count="6"/>
                 <test:non-blocking-processor/>
-            </foreach>
+            </until-successful>
+        </until-successful>
+        <test:assert expression="#[payload == 'Test Message']"/>
+    </flow>
+
+    <flow name="foreach_and_untilSuccessful" initialState="stopped">
+        <test:assert-intercepting responseSameTask="false"/>
+        <foreach>
+            <test:assert-intercepting responseSameTask="false" count="2"/>
+            <test:non-blocking-processor/>
+            <until-successful>
+                <test:assert-intercepting responseSameTask="false" count="6"/>
+                <test:non-blocking-processor/>
+            </until-successful>
         </foreach>
     </flow>
 
+    <flow name="untilSuccessful_and_foreach" initialState="stopped">
+        <test:assert-intercepting responseSameTask="false"/>
+        <set-payload value="#[mel: [ 'A', 'B' ]  ]"/>
```

Comment:
use DW instead of MEL. Now that you're at it, migrate all other MEL usages in this file to mel